appcues/src/main/java/com/appcues/ui/extensions/StyleComponentExt.kt

```diff
@@ -76,32 +100,50 @@ private fun FontFamily.Companion.getSystemFontFamily(fontName: String?): FontFam
     return null
 }
 
-// try to load a custom font from a resource in the host application
-private fun FontFamily.Companion.getFontResource(context: Context, fontName: String?): FontFamily? {
+private fun FontFamily.Companion.getFontResource(context: Context, packageNames: List<String>, fontName: String?): FontFamily? {
     // Font resources are only available on API 26+ (Android 8.0, "O"), so don't even try on lower versions
     if (VERSION.SDK_INT >= VERSION_CODES.O && fontName != null) {
-        // this is an attempt to provide convenience to convert to valid resource names, for ex. if a font
-        // name of "Lato-Black" was used, it will auto convert to "lato_black" resource name
-        val resourceName = fontName.lowercase().replace("-", "_")
-        val fontId = context.resources.getIdentifier(resourceName, "font", context.packageName)
-        if (fontId != 0) {
-            // Even if the font identifier is found, this typeface may not be available on the current
-            // OS level, if any api-specific resource folders have been used, i.e. "font-v30". We do
-            // another check here to try and catch this, since otherwise it will result in an uncaught
-            // exception during Compose render time. That exception will crash the app when
-            // trying to show the Appcues experience.
-            @Suppress("SwallowedException")
-            return try {
-                // the result of this getFont is unused, but success indicates the resource is available
-                // and we can progress with the creation of the Font and FontFamily
-                context.resources.getFont(fontId)
-                FontFamily(Font(fontId))
-            } catch (_: NotFoundException) {
-                null
-            } catch (_: NullPointerException) {
-                // this can happen in our snapshot unit tests when the font is not available
-                null
-            }
+        val mergedPackageNames = mutableSetOf<String>().apply {
+            add(context.packageName)
+            addAll(packageNames)
+        }
+
+        // iterate the packageNames list and return the first not null font
+        mergedPackageNames.forEach {
+            val font = getFontResource(context, it, fontName)
+            if (font != null) return font
+        }
+    }
+
+    // font was not found in resources
+    return null
+}
+
+// try to load a custom font from a resource in the host application
+@RequiresApi(VERSION_CODES.O)
+@SuppressLint("DiscouragedApi")
```

Comment:
curious what uses DiscouragedApi? was that there before?

Reply:
getIdentifier since it can lead to errors